### PR TITLE
fix(sts): harden MossFormer2SE local loading

### PR DESCRIPTION
**Pull request by beshkenadze**
## Summary
- fail early in `fromLocal` when no `.safetensors` are present
- reject duplicate keys in local weights after key normalization (`module.`/`mossformer.` sanitization)
- refactor loader path (`fromPretrained`/`fromLocal`) into shared config/weights/model build steps
- add integration tests for missing local safetensors and normalized duplicate keys

## Test Plan
- xcodebuild test -scheme MLXAudio-Package -testPlan MLXAudio-Package -destination 'platform=macOS,arch=arm64' -only-testing:'MLXAudioTests/MossFormer2SEIntegrationTests/fromLocalRejectsMissingSafetensors()' -only-testing:'MLXAudioTests/MossFormer2SEIntegrationTests/fromLocalRejectsNormalizedDuplicateWeightKeys()'

### CODE REVIEW

File: Sources/MLXAudioSTS/Models/MossFormer2SE/MossFormer2Model.swift [hunk 1]
```diff
@@ -311,11 +359,27 @@ public final class MossFormer2SEModel {
...
+    }
+
+    public static func fromLocal(_ directory: URL) throws -> MossFormer2SEModel {
+        let config = try loadConfig(from: directory, fallbackPolicy: .fallbackOnlyWhenMissing)
+        let weights = try loadWeights(from: directory, duplicateKeyPolicy: .fail)
+        return try buildModel(config: config, weights: weights)
+    }
+
+    public func enhance(_ audioInput: MLXArray, dither: Float = 0.0) throws -> MLXArray {
         guard audioInput.ndim == 1 else {
```
[the remark this answers is not in the document]
Comment (beshkenadze) [reply Sources/MLXAudioSTS/Models/MossFormer2SE/MossFormer2Model.swift:383]: This related to https://github.com/Blaizzy/mlx-audio-swift/pull/38

File: Sources/MLXAudioSTS/Models/MossFormer2SE/MossFormer2Model.swift [hunk 2]
```diff
@@ -257,50 +263,92 @@ public final class MossFormer2SEModel {
...
-            weights.merge(fileWeights) { _, new in new }
+            switch duplicateKeyPolicy {
+            case .overwrite:
+                weights.merge(fileWeights) { _, new in new }
+            case .fail:
+                var normalizedKeys = Set(weights.keys.map(normalizedWeightKey))
+                for (key, value) in fileWeights {
+                    let normalized = normalizedWeightKey(key)
+                    if normalizedKeys.contains(normalized) {
+                        throw MossFormer2SEError.duplicateWeightKey(normalized)
```
[the remark this answers is not in the document]
Comment (beshkenadze) [reply Sources/MLXAudioSTS/Models/MossFormer2SE/MossFormer2Model.swift]: Done in `a8b9ed3`: moved `normalizedKeys` set initialization outside the shard loop and update it incrementally. This keeps behavior unchanged while avoiding per-file set rebuilds.

**Comment by beshkenadze** [comment]
Superseded by a combined PR with all MossFormer2 STS fixes (dither, local loading, Kaldi fbank, mel norm, RoPE).
